pandas/io/excel/_openpyxl.py

```diff
@@ -484,7 +489,11 @@ def _write_cells(
                 row=freeze_panes[0] + 1, column=freeze_panes[1] + 1
             )
 
+        notes_col = None
+
         for cell in cells:
+            if notes_col is None:
+                notes_col = startcol + cell.col + 1
```

Comment:
Can you instead do e.g. `next(iter(cells)).col` and take this out of the for loop?

Reply:
I'm not getting this one, `notes_col` is the 1st column where data is inserted (to skip over all index columns) since when writing to excel I need the actual cell column number (in the excel sheet).

Reply:
My understanding is that `notes_col is None` is only true on the first pass of this for loop, and so this is just a way of getting the first cell in `cells`. The code `next(iter(cells))` also gives you the first cell. The full code would be:

    notes_col = start_col + next(iter(cells)).col + 1

I think this can go below the `if notes is None or notes_col is None:` as we don't need it unless we get passed that.

Reply:
got it, my problem was that cells is already an iterator, `next(cells)` is enough. This is fixed now.

Reply:
We opted with doing

```
if notes and cells:
         notes_col = startcol + next(cells).col + 1
```

This works because cells is already an Iterable[ExcelCell] so there's no need for iter(cells).
Plus there's also no need to check if notes_col is None since the checks comes right after the initialisation of notes_col.